core/src/main/java/com/linecorp/armeria/client/HttpResponseDecoder.java

```diff
@@ -286,8 +286,15 @@ private void close(@Nullable Throwable cause,
             if (cancelTimeout()) {
                 actionOnTimeoutCancelled.accept(cause);
             } else {
-                if (cause != null && !Exceptions.isExpected(cause)) {
-                    logger.warn("Unexpected exception:", cause);
+                if (cause != null && logger.isWarnEnabled() && !Exceptions.isExpected(cause)) {
+                    StringBuilder logMsg = new StringBuilder("Unexpected exception while closing");
+                    if (request != null) {
+                        logMsg.append(" request to ").append(request.authority());
+                    }
+                    if (cause instanceof ResponseTimeoutException) {
+                        logMsg.append(" after ").append(responseTimeoutMillis).append("ms");
+                    }
```

Comment:
nit: `logMsg.append(':')`, here, will make it prettier, perhaps?

Reply:
I don't think so, there is going to be a newline right after anyways (yaml log? :D)